frontend/src/components/common/Error/ErrorFallback.tsx

```diff
@@ -0,0 +1,37 @@
+import { HTTP_ERROR_MESSAGE } from '@/constants/api';
+import { isHTTPErrorStatus } from '@/utils/isHTTPErrorStatus';
+import Button from '../Button/Button';
+import * as S from './ErrorFallback.styled';
+
+export interface ErrorFallbackProps {
+  statusCode: number;
+  resetError?: () => void;
+}
+
+const ErrorFallback = ({ statusCode, resetError }: ErrorFallbackProps) => {
+  const currentStatusCode = statusCode;
+
+  return (
+    <S.Container>
+      <S.Wrapper>
+        {isHTTPErrorStatus(currentStatusCode)
+          ? HTTP_ERROR_MESSAGE[currentStatusCode].HEADING
+          : 'Unknown Error'}
+      </S.Wrapper>
+      <S.Wrapper>
+        {isHTTPErrorStatus(currentStatusCode)
+          ? HTTP_ERROR_MESSAGE[currentStatusCode].BODY
+          : '알 수 없는 오류가 발생했습니다.'}
+      </S.Wrapper>
+      <S.Wrapper>
+        <Button onClick={resetError}>
+          {isHTTPErrorStatus(currentStatusCode)
+            ? HTTP_ERROR_MESSAGE[currentStatusCode].BUTTON
+            : '홈으로 돌아가기'}
+        </Button>
+      </S.Wrapper>
+    </S.Container>
+  );
```

Comment:
코드 개선 아이디어 공유드립니다 ㅎㅎ
```suggestion

  const { heading, body, button } = generateHTTPErrorMessage(currentStatusCode);

  return (
    <S.Container>
      <S.Wrapper>{heading}</S.Wrapper>
      <S.Wrapper>{body}</S.Wrapper>
      <S.Wrapper>
          <Button onClick={resetError}>{button}</Button>
      </S.Wrapper>
    </S.Container>
  );
```
```ts
const generatHTTPErrorMessage = (statusCode: number): httpErrorMessage => {
    const isHTTPError = isHTTPErrorStatus(statusCode);

    const errorMessage = HTTP_ERROR_MESSAGE[statusCode]

    return errorMessage ?? HTTP_ERROR_MESSAGE.unknown;
}
```

Reply:
짚어주셔서 감사해요! 덕분에 더 간결하게 나타내볼수 있었습니다 ㅎ

Reply:
오 더 간결하네요!!